agent/app/agent_test.go

```diff
@@ -1473,3 +1533,55 @@ func newTestDataClient(t *testing.T) (data.Client, func()) {
 	}
 	return testClient, cleanup
 }
+
+func TestWaitUntilInstanceInService(t *testing.T) {
```

Comment:
could you make this and below 3 tests table driven? 
https://github.com/golang/go/wiki/TableDrivenTests